LanguageFeatures/Augmentation-libraries/augmented_expression_A04_t20_lib.dart

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The exact result of an `augmented` expression depends on what is
+/// being augmented, but it generally follows the same rules as any normal
+/// identifier:
+/// ...
+/// - Augmenting functions: When augmenting a function, `augmented` refers to
+///   the augmented function. Tear offs are not allowed, so this function must
+///   immediately be invoked.
+///
+/// @description Checks that it is a compile-time error to use type named
+/// `augmented` in an augmenting function body.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmented_expression_A04_t20.dart';
+
+augment void topLevelFunction() {
+  augmented? x;
```

Comment:
Yes, in this test it's perfectly well justified! All usages of the word `augmented` as a type should be flagged as an error in here.

Also, this is the place where `(augmented? x) {}` fits.

I suspect line 26 is perfect here, but shouldn't occur in any of those several other tests where it does appear (because those other tests are about other things, if I understand their description correctly).

Reply:
(All fixed in description)